cwe_checker_rs/src/checkers/cwe_78/context/mod.rs

```diff
@@ -0,0 +1,563 @@
+                new_state.set_expression_taint_and_store_constants(
+                    &def.tid,
+                    var,
+                    input,
+                    &self.project.stack_pointer_register,
+                )
+            }
+        }
+
+        new_state
+    }
+
+    /// Handles store definition updates
+    pub fn handle_store(
+        &self,
+        state: State,
+        def: &Term<Def>,
+        address: &Expression,
+        value: &Expression,
+    ) -> State {
+        let mut new_state = state;
+        let target = match address {
+            Expression::Var(ram_target) => ram_target,
+            Expression::Subpiece { arg, .. } => {
+                let argument: &Expression = arg;
+                match argument {
+                    Expression::Var(ram_target) => ram_target,
+                    _ => panic!("Unexpected target format for Store instruction!"),
+                }
+            }
+            _ => panic!("Unexpected target format for Store instruction!"),
+        };
+        new_state.taint_value_to_be_stored(&def.tid, target, value);
+
+        new_state
+    }
+
+    /// Gets the BlkEnd node of an external function call
+    pub fn get_source_node(&self, call_source: &Tid) -> NodeIndex {
+        let blk_end_node_id = self
+            .jmp_to_blk_end_node_map
+            .get(&(call_source.clone(), self.current_sub.unwrap().tid.clone()));
+
+        if let Some(blk_end_node) = blk_end_node_id {
+            *blk_end_node
+        } else {
+            panic!("Malformed Control Flow Graph.");
+        }
+    }
+}
+
+impl<'a> crate::analysis::backward_interprocedural_fixpoint::Context<'a> for Context<'a> {
+    type Value = State;
+
+    /// Get the underlying graph of the fixpoint computation
+    fn get_graph(&self) -> &Graph<'a> {
+        &self.graph
+    }
+
+    /// Merge two states
+    fn merge(&self, state1: &State, state2: &State) -> State {
+        state1.merge(state2)
+    }
+
+    /// Updates State according to side effects of the definition
+    fn update_def(&self, state: &State, def: &Term<Def>) -> Option<State> {
+        if state.is_empty() {
+            // Without taint there is nothing to propagate.
+            return None;
+        }
+        let mut new_state = state.clone();
+        // Check whether the def is the last def of a block and if so, create the
+        // Def Pi Map
+        if let Some(blk_start_node) = self.get_blk_start_node_if_last_def(def) {
+            new_state.set_pi_def_map(self.create_pi_def_map(blk_start_node));
+        }
+
+        match &def.term {
+            Def::Assign { var, value: input }
+            | Def::Load {
+                var,
+                address: input,
+            } => new_state = self.handle_assign_and_load(new_state, def, var, input),
+            Def::Store { address, value } => {
+                new_state = self.handle_store(new_state, def, address, value)
+            }
+        }
+
+        Some(new_state)
```

Comment:
I think we should also check whether the `def` is the first node of the block. In that case we could remove the `pi_def_map` from the `new_state` and thus preventing unnecessary information to be stored in the computation struct.